docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+
+![Enable Socket Mode](./images/enable-sockets.png)
+
+When you enable Socket Mode, Slack gives you an application token that you will need when you start the app.
+This token will typically start with `xapp`.
+It is located under App-Level Tokens in Basic Information.
+
+### Create a Slash Command
+
+Select "Slash Commands" in the sidebar navigation, and click "Create New Command".
+Add a new command for `/workqueue`.
+Since you have Socket Mode enabled, you won't need to add a request URL.
+
+![Create a Slash Command](./images/enable-slash-command.png)
+
+### Subscribe to Events
+
+Under Features in the navigation bar, select Event Subscriptions and ensure Enable Events is turned ON.
+The navigate to the section under "Subscribe to bot events" and add subscriptions for:
+
+- `message.groups`
+- `message.im`
+- `message.channels`
+
+![Subscribe to bot events](./images/subscribe-to-bot-events.png)
+
+Once you've done this, click "Save Changes"
+
+### Set OAuth & Permissions
+
+Under Features in the navigation bar, go to OAuth & Permissions.
+
+You'll need to enable the following Bot Token Scopes:
+
+- `channels:history`
+- `channels:read`
+- `chat:write`
+- `groups:history`
+- `groups:read`
+- `im:history`
+- `mpim:history`
+
+### Install App
+
+After you've done this, install the app in Slack by going to Install App under Settings in the navigation bar.
+
+### Get credentials
+
+You will need **Signing Secret** and **App Level Token** from the Basic Information page.
+And you will need the **Bot User OAuth Token** from the Install App page, which is also available on the OAuth & Permissions page.
+
+## Create TypeScript projects for the Slack App
+
+You need two TypeScript projects for this Slack application.
+You may have already created the first one for the Temporal Application in the [Prerequisite steps](/getting_started/typescript/dev_environment/).
+Here, make sure you have the second one for the Slack bot logic.
+
+This tutorial assumes that your TypeScript projects will be organized like this:
+
+```plaintext
+-- your-workqueue-slack-app
+  |-- temporal-application # This is the Temporal Application project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- workflows
+  |   |   |   |-- workqueue.ts
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- bot # This is the Slack bot project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- common-types # These types are shared by both projects
+      |-- types.ts
+```
+
+This tutorial uses `npm` as the package manager.
+
+```
+npm init
+```
+
+And ensures there is a `tsconfig.json` file in each project root.
+
+```
+tsc --init
+```
+
+### Dependencies for the projects
+
+**Temporal Application dependencies**
+
+In the `temporal-application` project, you will need the following dependencies in your `package.json`:
```

Comment:
I'd recommend removing these `package.json` snippets and replacing them with the relevant `npm i` commands to add the necessary packages, since this tutorial is meant to produce an end-to-end result and we're not cloning anything down. Stick to what's necessary (if anything) after the `npx` install route, that way we can cut this section down and make it more linear.